filemage/tests/test_filemage.py

```diff
@@ -0,0 +1,316 @@
+        raise NotImplementedError()
+
+    def cpu_affinity(self, cpus=None):
+        raise NotImplementedError()
+
+    def cpu_num(self):
+        raise NotImplementedError()
+
+    def environ(self):
+        raise NotImplementedError()
+
+    def num_handles(self):
+        raise NotImplementedError()
+
+    def num_ctx_switches(self):
+        raise NotImplementedError()
+
+    def num_threads(self):
+        raise NotImplementedError()
+
+    def threads(self):
+        raise NotImplementedError()
+
+    def children(self, recursive=False):
+        raise NotImplementedError()
+
+    def cpu_percent(self, interval=None):
+        raise NotImplementedError()
+
+    def cpu_times(self):
+        raise NotImplementedError()
+
+    def memory_info(self):
+        raise NotImplementedError()
+
+    def memory_full_info(self):
+        raise NotImplementedError()
+
+    def memory_percent(self, memtype="rss"):
+        raise NotImplementedError()
+
+    def memory_maps(self, grouped=True):
+        raise NotImplementedError()
+
+    def open_files(self):
+        raise NotImplementedError()
+
+    def connections(self, kind='inet'):
+        raise NotImplementedError()
+
+    def send_signal(self, sig):
+        raise NotImplementedError()
+
+    def suspend(self):
+        raise NotImplementedError()
+
+    def resume(self):
+        raise NotImplementedError()
+
+    def terminate(self):
+        raise NotImplementedError()
+
+    def kill(self):
+        raise NotImplementedError()
+
+    def wait(self, timeout=None):
+        raise NotImplementedError()
+
+
+def mockPrcoessIterServicesDown(attrs=None, ad_value=None):
+    def procIterFormat(proc):
+        proc.info = proc.as_dict(attrs=attrs, ad_value=ad_value)
+        return proc
+
+    yield procIterFormat(MockProcess(name='systemd', cmdline=['/sbin/init']))
+
+
+def mockPrcoessIterServicesUp(attrs=None, ad_value=None):
+    def procIterFormat(proc):
+        proc.info = proc.as_dict(attrs=attrs, ad_value=ad_value)
+        return proc
+
+    yield procIterFormat(MockProcess(name='systemd', cmdline=['/sbin/init']))
+    yield procIterFormat(
+        MockProcess(
+            name='postgres',
+            cmdline=[
+                '/usr/lib/postgresql/12/bin/postgres',
+                '-D',
+                '/var/lib/postgresql/12/main',
+                '-c',
+                'config_file=/etc/postgresql/12/main/postgresql.conf',
+            ],
+        )
+    )
+    yield procIterFormat(
+        MockProcess(
+            name='gateway',
+            cmdline=[
+                '/opt/filemage/bin/gateway',
+                '-c',
+                '/etc/filemage/config.yml',
+            ],
+        )
+    )
+
+
+def test_check(dd_run_check, aggregator, instance):
+    check = FilemageCheck('filemage', {}, [MOCK_INSTANCE])
+    dd_run_check(check)
+
+    aggregator.assert_all_metrics_covered()
+    aggregator.assert_metrics_using_metadata(get_metadata_metrics())
```

Comment:
These assertions (and therefore the whole test) are not really doing anything, because under the given conditions the check is not producing metrics. For `assert_all_metrics_covered` to be useful, you would have to assert individual metrics with [`aggregator.assert_metric`](https://datadoghq.dev/integrations-core/base/api/#datadog_checks.base.stubs.aggregator.AggregatorStub.assert_metric), as this asserts that all metrics emitted during the check run have been asserted with that method.

You'd need appropriate mocking for the check to generate metrics during the test, and assertions on them.

Reply:
resolved in 743314bd136fa41a564d99d590471733f4969a6b